SmartDeviceLink-iOS/SmartDeviceLink/SDLStreamingMediaManager.m

```diff
@@ -229,83 +251,52 @@ - (BOOL)sdl_configureVideoEncoderWithError:(NSError *__autoreleasing *)error {
 
     if (status != noErr) {
         // TODO: Log the error
-        if (*error != nil) {
+        if (*error == nil) {
             *error = [NSError errorWithDomain:SDLErrorDomainStreamingMediaVideo code:SDLStreamingVideoErrorConfigurationCompressionSessionCreationFailure userInfo:@{ @"OSStatus" : @(status) }];
         }
 
         return NO;
     }
-
-    // Set the bitrate of our video compression
-    int bitRate = 5000;
-    CFNumberRef bitRateNumRef = CFNumberCreate(NULL, kCFNumberSInt32Type, &bitRate);
-    if (bitRateNumRef == NULL) {
-        // TODO: Log & end session
-        if (*error != nil) {
-            *error = [NSError errorWithDomain:SDLErrorDomainStreamingMediaVideo code:SDLStreamingVideoErrorConfigurationAllocationFailure userInfo:nil];
-        }
-
-        return NO;
+    
+    if (self.videoEncoderSettings == nil) {
```

Comment:
If this is how it's being done, then the video encoder settings dict should be null_resettable, not nullable, and the api should be redesigned so that it's initialized on `init`, maybe an `initWithEncoderSettings` if possible. For testability, as many dependencies as possible should be injected at init time.